enhancements/api-review/custom-tags-aws.md

```diff
@@ -0,0 +1,144 @@
+   prohibit that later, though.
+
+## Proposal
+
+Add a new field `userTags` to `.spec.aws` of the `infrastructure.config.openshift.io` type. Tags included in the
+`userTags` field will be applied to new resources created for the cluster.
+
+Installer will apply these tags to all AWS resources it creates with terraform (e.g. bootstrap and master EC2 instances)
+
+All operators that create AWS resources (ingress, cloud credential, storage, image registry, machine-api)
+will apply these tags to all AWS resources they create.
+
+### User Stories
+
+SDE-1146 - IAM users and roles can only operate on resources with specific tags
+As a security-conscious ROSA customer, I want to restrict the permissions granted to Red Hat in my AWS account by using
+AWS resource tags.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Some components, like machine-api and image registry, already have some support to allow adminstrators to
+push tags into EC2 instance and S3 buckets, respectively.
+
+In the case of machine-api, a user has to specify tags on the MachineSet as part of the Day 2 operation.
+
+In the case of image registry, a user has to pre-configure an S3 bucket with the required tags, and use the
+image registry's bring-your-own-bucket feature.
+
+### Risks and Mitigations
+
+To reduce initial scope, tags are applied only at creation time and not reconciled. This behavior is atypical in
+Kubernetes and tends to cause confusion for users.
+
+## Design Details
+
+### Open Questions [optional]
+
+ - Since machine-api and image registry already have some support to allow adminstrators to
+   push tags into EC2 instance and S3 buckets, respectively, should these be out of scope? Or can we agree
+   that it's a nicer user experience to specify the tags one time, at install time, and have everything just
+   work after that. Our intention is to change the contract of user tags to state that all resources created
+   for the cluster are tagged with the user-provided tags, putting the user-provided tags on equal footing with
+   the kubernetes owned tag. If we have to carve out exceptions from there that the user needs to remember to
+   fill out themselves, then it dilutes that contract.
+
+ - How are upgrades handled? Is this an install-time only feature that requires a fresh cluster to adopt, or
+   can we somehow enable upgrade users to adopt it?
+
+### Test Plan
+
+TBD
+
+### Graduation Criteria
+
+TBD
+
+### Upgrade / Downgrade Strategy
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to make use of the enhancement?
```

Comment:
given the lack of reconciliation support, the general behavior (independent of upgrade) is:

if the tags config field is set (and it can only be set after upgrading to a sufficient ocp version since the api won't exist prior to that point in time), any components which create new aws resources after the config field is set would be expected to apply the tags.  Any aws resources that were already created would not get tags added.